src/sympnets/Sampling.py

```diff
@@ -256,6 +254,7 @@ def NUTS(net, N):
             call_lf = 0
             counter_lf = 0
 
+        r_sto = np.zeros(int(int(input_dim1*2)/2))
```

Comment:
This is what we require. In case, the first sample is rejected, it should now not cause an issue. I assume you have verified the dimensionality of `r_sto`